src/tvheadend/HTSPDemuxer.cpp

```diff
@@ -272,41 +273,43 @@ PVR_ERROR HTSPDemuxer::CurrentDescrambleInfo ( PVR_DESCRAMBLE_INFO *info )
   return PVR_ERROR_NO_ERROR;
 }
 
-int64_t HTSPDemuxer::GetTimeshiftTime() const
-{
-  CLockObject lock(m_mutex);
-  return m_timeshiftStatus.shift;
-}
-
-int64_t HTSPDemuxer::GetTimeshiftBufferStart() const
+bool HTSPDemuxer::IsTimeShifting() const
 {
-  CLockObject lock(m_mutex);
+  if (!m_subscription.IsActive())
+    return false;
 
-  // Note: start/end mismatch is not a bug. tvh uses inversed naming logic here!
-  return m_timeshiftStatus.end;
-}
+  if (m_subscription.GetSpeed() != SPEED_NORMAL)
+    return true;
 
-int64_t HTSPDemuxer::GetTimeshiftBufferEnd() const
-{
   CLockObject lock(m_mutex);
+  if (m_timeshiftStatus.shift != 0)
+    return true;
 
-  // Note: start/end mismatch is not a bug. tvh uses inversed naming logic here!
-  return m_timeshiftStatus.start;
+  return false;
 }
 
-bool HTSPDemuxer::IsTimeShifting() const
+bool HTSPDemuxer::IsRealTimeStream() const
 {
   if (!m_subscription.IsActive())
     return false;
 
-  if (m_subscription.GetSpeed() != SPEED_NORMAL)
-    return true;
+  /* Handle as real time when reading close to the EOF (10 secs) */
+  CLockObject lock(m_mutex);
+  return (m_timeshiftStatus.shift < 10000000);
+}
+
+PVR_ERROR HTSPDemuxer::GetStreamTimes(PVR_STREAM_TIMES *times) const
+{
+  *times = {0};
 
   CLockObject lock(m_mutex);
-  if (m_timeshiftStatus.shift != 0)
-    return true;
 
-  return false;
+  times->startTime = m_startTime;
+  times->ptsStart = 0;
+  times->ptsBegin = static_cast<int64_t>(static_cast<double>(m_timeshiftStatus.start / 1000000)) * DVD_TIME_BASE;
+  times->ptsEnd = static_cast<int64_t>(static_cast<double>(m_timeshiftStatus.end / 1000000)) * DVD_TIME_BASE;
```

Comment:
Remember the "Note: start/end mismatch is not a bug. tvh uses inversed naming logic here!" comment

Reply:
TBH, I never understood this. As a matter of fact, it's working now - it's carefully runtime tested. Too lazy to think through this, tbh. I take it like I observed it.

Reply:
If it works then it's all good, it could be that the hack we had in place is some old stuff that is no longer relevant in newer tvheadend versions.